system/security/shared/caching.cpp

```diff
@@ -759,14 +771,24 @@ void CPermissionsCache::managedFileScopesCacheFillThread()
 
 void CPermissionsCache::stopManagedFileScopeCacheFillThread()
 {
-    if (m_fileScopeCacheFillThread.joinable())
+    bool threadExited = false;
+    // If the fill thread is running, signal it to stop and wait for it to exit
+    if (m_fileScopeCacheReady)
```

Comment:
This variable is being used for two distinct purposes - is the cache ready, and is there a refresh thread running.